src/nock/matcher.ts

```diff
@@ -189,9 +185,8 @@ export class Matcher {
     // response
     if (oldTraffic.response !== undefined) {
       await this.matchResponse(
-        oldTraffic.response,
-        newTraffic.response,
-        contentEncoding
+        oldTraffic.decodedResponse ?? oldTraffic.response,
```

Comment:
Is there any chance that we end up matching decodedResponse (Record<string, any>) and response (Buffer)?

Reply:
If there is no Content-Encoding header set with `response` and still containing Buffer then yes, but I'm not sure if that can even happen looking at [nock parsing response](https://github.com/nock/nock/blob/2752313ca3cb0cb2aaf4ed4cab71b62e5c074333/lib/recorder.js#L27-L33)